src/components/application_manager/test/commands/mobile/show_constant_tbt_test.cc

```diff
@@ -0,0 +1,381 @@
+    expected_message_ = CreateExpectedMessage();
+    mock_app_ = CreateMockApp();
+    ON_CALL(app_mngr_, application(kConnectionKey))
+        .WillByDefault(Return(mock_app_));
+    ON_CALL(app_mngr_, GetPolicyHandler())
+        .WillByDefault(ReturnRef(mock_polici_handler_interface_));
+  }
+
+ protected:
+  MessageSharedPtr message_;
+  SharedPtr<ShowConstantTBTRequest> request_;
+  MessageSharedPtr expected_message_;
+  MockAppPtr mock_app_;
+  MockPolicyHandlerInterface mock_polici_handler_interface_;
+};
+
+TEST_F(ShowConstantTBTRequestTest, Run_AllCorrectData_SendHMIRequest) {
+  EXPECT_CALL(mock_message_helper_, ProcessSoftButtons(_, _, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(mock_message_helper_, VerifyImage(_, _, _))
+      .Times(2)
+      .WillRepeatedly(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId));
+
+  EXPECT_CALL(mock_message_helper_, SubscribeApplicationToSoftButton(_, _, _));
+
+  smart_objects::SmartObject msg_params;
+  EXPECT_CALL(*mock_app_, set_tbt_show_command(_))
+      .WillOnce(SaveArg<0>(&msg_params));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  if (request_->Init()) {
+    request_->Run();
+  }
+
+  EXPECT_TRUE((*expected_message_)[am::strings::msg_params]
+                                  [am::hmi_request::navi_texts] ==
+              (*result)[am::strings::msg_params][am::hmi_request::navi_texts]);
+}
+
+TEST_F(ShowConstantTBTRequestTest,
+       Run_NonExistentApp_ApplicationNotRegistered) {
+  MockAppPtr mock_app_nonexistent;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_nonexistent));
+
+  ExpectedManageMobileCommandResultCode(
+      mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+
+  if (request_->Init()) {
+    request_->Run();
+  }
+}
+
+TEST_F(ShowConstantTBTRequestTest, Run_EmptyParams_InvalidData) {
+  if (message_->keyExists(am::strings::msg_params)) {
+    message_->erase(am::strings::msg_params);
+  }
+
+  ExpectedManageMobileCommandResultCode(mobile_apis::Result::INVALID_DATA);
+
+  request_ = CreateCommand<ShowConstantTBTRequest>(message_);
+  if (request_->Init()) {
+    request_->Run();
+  }
+}
+
+TEST_F(ShowConstantTBTRequestTest, Run_ProcessSoftButtonsInvalid_InvalidData) {
+  mobile_apis::Result::eType invalid_data = mobile_apis::Result::INVALID_DATA;
+  EXPECT_CALL(mock_message_helper_, ProcessSoftButtons(_, _, _, _))
+      .WillOnce(Return(invalid_data));
+
+  ExpectedManageMobileCommandResultCode(invalid_data);
+
+  if (request_->Init()) {
+    request_->Run();
+  }
+}
+
+TEST_F(ShowConstantTBTRequestTest, Run_TurnIconInvalid_InvalidData) {
+  EXPECT_CALL(mock_message_helper_, ProcessSoftButtons(_, _, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  mobile_apis::Result::eType invalid_data = mobile_apis::Result::INVALID_DATA;
```

Comment:
@BSolonenko better to give `expected_result_code` name for such variables

Reply:
@AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/1977/commits/2127b6028b54933a3f410a26f186212ea4845467